### PR TITLE
Call to login

### PR DESCRIPTION
**Pull request by eodero**

### CODE REVIEW

File: src/modules/login/components/Login.js [hunk 1]
```diff
@@ -1,24 +1,90 @@
...
+import ClipLoader from 'react-spinners/ClipLoader'
+import { useNavigate } from 'react-router-dom'
+import axios from 'axios'
+import { homePath } from '../../home/routes/HomeRoute'
 
 export const Login = () => {
+    const [isPending, setIsPending] = useState(false)
+    const [error, setError] = useState(null)
+    const navigate = useNavigate()
+    const [loginData, SetLoginData] = useState({
```
Comment (Leigh-K) [inline src/modules/login/components/Login.js]:
```suggestion
    const [loginData, setLoginData] = useState({
```

File: src/modules/login/components/Login.js [hunk 2]
```diff
@@ -1,24 +1,90 @@
...
+            return {
+                ...prevloginData,
+                [e.target.name]: e.target.value,
+            }
+        })
+    }
+
+    const handleSubmit = async (e) => {
+        e.preventDefault()
+
```
Comment (Leigh-K) [inline src/modules/login/components/Login.js:20]: set loading state here

File: src/modules/login/components/Login.js [hunk 3]
```diff
@@ -1,24 +1,90 @@
-import React from 'react'
+import React, { useState } from 'react'
 import styles from '../styles/Login.module.css'
+import ClipLoader from 'react-spinners/ClipLoader'
+import { useNavigate } from 'react-router-dom'
+import axios from 'axios'
+import { homePath } from '../../home/routes/HomeRoute'
 
 export const Login = () => {
+    const [isPending, setIsPending] = useState(false)
```
Comment (Leigh-K) [inline src/modules/login/components/Login.js]:
```suggestion
    const [isloading, setIsLoading] = useState(false)
```

File: src/modules/login/components/Login.js (hunk 2 above)
Comment (eodero) [reply src/modules/login/components/Login.js:20]: - changes completed

File: src/modules/login/components/Login.js (hunk 1 above)
Comment (eodero) [reply src/modules/login/components/Login.js]: changes completed

File: src/modules/home/components/LoginButton.js [hunk 4]
```diff
@@ -5,14 +5,11 @@ import { loginPath } from '../../login/routes/LoginRoute'
...
     const navigate = useNavigate()
-
     const onClick = () => navigate(loginPath)
 
     return (
-        <div>
-            <button className={styles.loginButton} onClick={onClick}>
-                Log in
-            </button>
+        <div onClick={onClick}>
```
Comment (Leigh-K) [inline src/modules/home/components/LoginButton.js]: the on click was on the button previously, move it back there

File: src/modules/login/styles/Login.module.css [hunk 5]
```diff
@@ -50,3 +55,12 @@
...
 }
+
+.button:disabled {
+    opacity: 0.5;
+}
+
+.spinner {
+    animation: spin infinite 5s linear;
+    color: #ffffff;
+}
```
Comment (Leigh-K) [inline src/modules/login/styles/Login.module.css]: unused style remove

File: src/modules/login/components/Login.js [hunk 6]
```diff
@@ -1,24 +1,76 @@
...
+                setIsLoading(false)
+                navigate(homePath)
+            }
+        } catch (err) {
+            if (err.name === 'AxiosError') {
+                console.log(err.message)
+            } else {
+                setError('Login unsuccessful')
+            }
+        }
```
Comment (Leigh-K) [inline src/modules/login/components/Login.js:36]: add in a finally block here (after the catch) to set loading to false
```
finally {
setIsLoading(false)
}
```

Comment (Leigh-K) [inline src/modules/login/components/Login.js]: this whoile catch block could just be
```
 setError('Login unsuccessful')
 ```
 No reason to specifically check an axios error or anything

**Review by Leigh-K — APPROVED** [review]

**Pull request merged**